src/index.js

```diff
@@ -38,15 +38,29 @@ function startExperiment(expId) {
   }
 }
 
-document.addEventListener('DOMContentLoaded', () => {
-  if (config) {
-    Share.listen(config, () => {
-      START_EXPERIMENT();
-      setupToolbar();
-      setupEditButton();
-    });
+// listen to session iframe
+window.addEventListener('message', e => {
+  if (e.data.type === 'preview') {
+    window.prismicSession = e.data.data
+    if (config) {
+      Share.listen(config, () => {
+        START_EXPERIMENT();
+        setupToolbar();
+        setupEditButton();
+      });
+    }
   }
-});
+})
+
+// session iframe
+const parser = document.createElement('a');
+parser.href = PRISMIC_ENDPOINT;
+const iframe = document.createElement('iframe');
+iframe.src = `${parser.protocol}//${parser.host}/previews/session/get`;
```

Comment:
In fact, instead of creating another iframe, we could use the one we use for display loading.